### ISSUE TITLE
Handling of unparsable header fields (unencoded UTF-8 subject)

### ISSUE THREAD

**Issue by RST-J** [mikel/mail#1012]
Hello,
we are using this great gem since quite a while and now stumbled upon a problem when fetching broken (i.e. non-rfc-compliant) mails.
An example mail is:
```
Return-Path: <bar@example.com>
X-Original-To: bar@example.com
Delivered-To: bar@example.com
From: The sender <foo@example.com>
To: bar@example.com
Subject: Non-escaped subject with eviäl umlaut
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8;
 format=flowed
Date: Fri, 27 May 2016 05:41:18 +0200
Message-ID:
 <aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee@mail.example.com>

Hello Mailbox!
```

What happens when dealing with this email, is that a message is written to Stderr:
> WARNING: Could not parse (and so ignoring) 'Subject: Non-escaped subject with evi�l Umlaut'

and `mail.subject` is `nil` - not surprising as it was said to be ignored.
I thought that I'd could find information about such problems in either `mail.errors` or `mail.header['Subject'].errors` but both are empty arrays although the doc indicates that this would be exactly their purpose.
I cannot even check for `nil` as subject value as a workaround as emails with empty subject also have `nil` there.
Anything I am doing wrong or is this maybe a bug?

Im using mail 2.6.4 with MRI 2.2.1

**Comment by RST-J** [comment]
I investigated the issue further and found out that this only happens within our Rails environment.
If I read the very same file with the very same checkout of 2.6.4 once in the rails console and once from a plain irb from the checkout directory then in the first case I get the warning and an empty subject. In the latter case there is no warning and the subject is 
``"Non-escaped subject with evi\xE4l umlaut"``

Does anyone have an idea what could be the cause for this? I already checked that neither `Encoding.default_internal` nor `Encoding.default_external` are the cause.
